integration-core/src/main/java/org/symphonyoss/integration/core/bootstrap/IntegrationBootstrapInfo.java

```diff
@@ -52,4 +42,15 @@ public Integration getIntegration() {
     return integration;
   }
 
+  public int getRetryAttemptCounter() {
+    return retryAttempts;
+  }
+
+  public int registerRetryAttempt() {
+    if (retryAttempts < Integer.MAX_VALUE) {
+      return ++retryAttempts;
```

Comment:
This variable might be an AtomicInteger to avoid concurrency issues. You can use incrementAndGet() to return the integer value.